### PR TITLE
Update distribution section

### PR DESCRIPTION
**Pull request by wllmwu**
### Administrative Info

Monday Board ID: 851212620
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes

- Implemented the new calendar component in the agency profile form
    - A popup appears when the user selects a custom distribution date, so they can set the start time or remove the date
    - User-selected dates also appear in a list next to the calendar, where they can also set start times
- Changed time string formats to use explicit time offset
    - All ISO 8601-formatted strings now look like: `YYYY-MM-DDThh:mm-07:00`
- Resolved *most* linting errors
    - Some errors remain in frontend/src/pages/Agency/Agency.js and frontend/src/components/AgencyTable/Selected.js related to `for...in` loops (I was not sure how to fix without breaking the loops)
    - Linter ignores build folder
    - Errors become just warnings when compiling in frontend

### Testing

How did you confirm your changes worked?

- Created new agencies in the database and updated some existing ones; time formats, selected dates, etc. were as expected
- Linter gives no errors besides the ones I left alone
- Tests pass

### Confirmation of Change

Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

Screen Shot 2021-05-05 at 10 17 07

### CODE REVIEW

File: frontend/src/pages/Agency/Agency.js [hunk 1]
```diff
@@ -224,6 +175,51 @@ function AgencyTable() {
     return false;
   }
 
+  /**
```
Comment (sblumin) [inline frontend/src/pages/Agency/Agency.js:178]: Why was this function moved down in the file?

**Review by sblumin — COMMENTED** [review]
Great job on this PR! Your updates to the distribution calendar and timing look great. Also, thanks so much for addressing most of the linting errors, much appreciated :) I made a couple comments so take a look and respond to those. Good work y'all

File: backend/routes/agency.js [hunk 2]
```diff
@@ -178,7 +180,7 @@ router.get("/:id", isAuthenticated, async (req, res, next) => {
  * @params - the object id of the Agency
  * @returns the fetched Agency object in Json
  */
-router.get("/table/all", async (req, res, next) => {
```
[the remark this answers is not in the document]
Comment (wllmwu) [reply backend/routes/agency.js:181]: They weren't being used, which caused eslint errors

File: frontend/src/App.js [hunk 3]
```diff
@@ -1,7 +1,8 @@
 import React, { useState } from "react";
+// eslint-disable-next-line no-unused-vars
```
[the remark this answers is not in the document]
Comment (wllmwu) [reply frontend/src/App.js:2]: The next line has some named imports which aren't used (`Router` and `Link` I think), and that also causes eslint errors - but removing those imports breaks the app (I guess they have to be ordered that way), so this comment makes eslint ignore that line

File: frontend/src/pages/Agency/Agency.js (hunk 1 above)
Comment (wllmwu) [reply frontend/src/pages/Agency/Agency.js:178]: More linting - eslint wanted functions to be defined in different orders in a few files (I think to make sure they are defined before used)

File: frontend/src/components/AgencyProfileForm/DistributionCalendar/TimeInputPopup.jsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+ * take a String
+ * - {Function} onDelete: callback for when the user clicks the delete button,
+ * should take no arguments
+ * - {Function} onHide: callback for when the user clicks outside of the
+ * TimeBox, should take no arguments
+ */
+class TimeInputPopup extends Component {
+  constructor(props) {
+    super(props);
+
```
[the remark this answers is not in the document]
Comment (wllmwu) [reply frontend/src/components/AgencyProfileForm/DistributionCalendar/TimeInputPopup.jsx]: Removed it!

**Review by sblumin — APPROVED** [review]
LGTM!

**Review by ryanbuui — APPROVED** [review]
LGTM! That calendar looks awesome. Sorry again for the delay!

**Pull request merged**